sdc/datatypes/hpat_pandas_series_rolling_functions.py

```diff
@@ -0,0 +1,145 @@
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+import numpy
+import pandas
+
+from numba.extending import register_jitable
+from numba.types import float64
+
+from sdc.datatypes.common_functions import TypeChecker
+from sdc.datatypes.hpat_pandas_series_rolling_types import SeriesRollingType
+from sdc.utils import sdc_overload_method
+
+
+@register_jitable
+def arr_finite_min(arr):
+    """Calculate minimum of finite values"""
+    finite_arr = arr[numpy.isfinite(arr)]
+    if len(finite_arr) == 0:
+        return numpy.nan
+
+    return finite_arr.min()
+
+
+def gen_hpat_pandas_series_rolling_impl(rolling_func, output_type=None):
+    """Generate series rolling methods implementations based on input func"""
+    nan_out_type = output_type is None
+
+    def impl(self):
+        win = self._window
+        minp = self._min_periods
+
+        input_series = self._data
+        input_arr = input_series._data
+        length = len(input_arr)
+        out_type = input_arr.dtype if nan_out_type == True else output_type # noqa
+        output_arr = numpy.empty(length, dtype=out_type)
+
+        for i in range(min(win, length)):
+            arr_range = input_arr[:i + 1]
+            finite_values_num = numpy.isfinite(arr_range).sum()
+            if finite_values_num < minp:
+                output_arr[i] = numpy.nan
+            else:
+                output_arr[i] = rolling_func(arr_range)
+
+        for i in range(min(win, length), length):
+            arr_range = input_arr[i + 1 - win:i + 1]
+            finite_values_num = numpy.isfinite(arr_range).sum()
+            if finite_values_num < minp:
```

Comment:
Same question as above - do we need this check?

Reply:
According the documentation on `rolling`, I think so.